app/api/qstash/workers/sync-users/route.ts

```diff
@@ -0,0 +1,104 @@
+// This is a public API endpoint that will be invoked by QStash.
+import { verifySignatureAppRouter } from "@upstash/qstash/dist/nextjs";
+import { NextRequest, NextResponse } from "next/server";
+import { loadQstash } from "../../../../../lib/qstash";
+import { UsersSyncPayload } from "../../../../../lib/qstash/types";
+import prisma from "../../../../../lib/prisma";
+import {
+  fetchFarcasterProfiles,
+  UserProfile,
+} from "../../../../../lib/airstack";
+
+async function handler(request: NextRequest) {
+  // Get the payload from the request and extract the offset
+  const body: UsersSyncPayload = await request.json();
+  const { offset } = body;
+
+  // Declaring the batch size
+  const batchSize = 400;
+
+  try {
+    // Read users from the WaitlistedUser Model in the database, grouped by fid with and taken with an offset
+    const usersBatch = await prisma.waitlistedUser.findMany({
+      distinct: ["fid"],
+      skip: offset,
+      take: batchSize,
+      orderBy: { fid: "asc" },
+    });
+
+    if (usersBatch.length > 0) {
+      // Transforming the fids number array to a string array to satisfy the Airstack query
+      const fidsString = usersBatch.map((fid) => fid.toString());
+
+      // Call Airstack to fetch all useful information about the users in the batch
+      const usersToUpdate: UserProfile[] = [];
+      let pointer = "";
+
+      do {
+        const res = await fetchFarcasterProfiles(fidsString, pointer);
+        if (res) {
+          const { profiles, pageInfo } = res;
+          usersToUpdate.push(...profiles);
+          pointer = pageInfo.nextCursor;
+        }
+      } while (pointer);
+
+      // Update the users in the database with the new information
+      // Maybe there's a better way to do this to avoid doing N queries to the DB
+      for (const user of usersToUpdate) {
```

Comment:
Instead of this here, we should do something like:

1. Divide usersToUpdate in chunks of 200 users
2. Run the updates in parallel using: `await Promise.all(usersToUpdateChunk.map(async (chunk) => prisma.waitlistedUser.updateMany(...)))`